data-table/__tests__/table.js

```diff
@@ -4,6 +4,113 @@ beforeEach(() => {
     document.body.innerHTML = '<div id="div-id"></div>'
 });
 
+describe('basic tests to ensure createDataTable can function well', () => {
+    beforeEach(() => {
+        table.createDataTable({
+            'wrapperDivId': 'div-id'
+        });
+    });
+
+    test('check the add column button', () => {
+        const config = {
+            'wrapperDivId': 'div-id',
+        };
+        table.createDataTable(config);
+        const contents = document.getElementById(config.wrapperDivId).textContent;
+        expect(contents.substr(0, 12)).toEqual("+ Add column");
+    });
+
+    test('check the default value of rows', () => {
+        const contents = document.getElementsByTagName('table')[0].rows.length;
+        expect(contents).toEqual(4);
+    });
+
+    test('check the default value of columns', () => {
+        const contents = document.getElementsByTagName('table')[0].rows[0].cells.length;
+        expect(contents).toEqual(4);
+    });
+
+    test('properly test the first cell', () => {
+        const contents = document.getElementsByClassName('data-table-cell')[0].textContent;
+        expect(contents).toEqual('Rows');
+    });
+
+    test('properly test the first cell', () => {
+        const contents = document.getElementsByClassName('data-table-cell')[1].textContent;
+        expect(contents).toEqual('Column 1');
+    });
+
+    test('check that subDivs are created', () => {
+        expect(document.getElementsByClassName("SubDiv")).not.toEqual(null);
+    });
+
+    test('check that data-table element is created', () => {
+        expect(document.getElementsByClassName("data-table")).not.toEqual(null);
+    });
+});
+
+describe('basic tests to ensure the buttons can function well', () => {
+    beforeEach(() => {
+        table.createDataTable({
+            'wrapperDivId': 'div-id'
+        });
+    });
+
+    test('check the add column button fuctionality', () => {
+        const addButton = document.getElementsByClassName("add-row-button");
+        let contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(16);
+        addButton[0].click();
+        contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(20);
+    });
+
+    test('check the delete column button fuctionality', () => {
+        const addButton = document.getElementsByClassName("add-row-button");
+        let contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(16);
+        addButton[1].click();
+        contents = document.getElementsByClassName("data-table-cell");
+        expect(contents.length).toEqual(12);
```

Comment:
`16` and `12` are magic numbers - can you describe them? Consider something like (pseudocode):
```
const numRows = 4;
const numCols = 4;
expect(contents.length).toEqual(numRows * numCols);
click;
expect(contents.length).toEqual((numRows-1) * numCols);
```